include/deal.II/multigrid/mg_transfer_matrix_free.h

```diff
@@ -83,8 +84,15 @@ class MGTransferMatrixFree : public MGLevelGlobalTransfer<LinearAlgebra::distrib
 
   /**
    * Actually build the information for the prolongation for each level.
+   *
+   * If @transfer_solution_vectors is true, the internal data structures
+   * are set-up for transfer of solution vectors instead of residuals.
+   * The former is relevant when one needs to have a representation of
+   * solution vector on each multi-grid level, whereas the latter is
+   * used to implement geometric-multigrid preconditioning.
    */
-  void build (const DoFHandler<dim,dim> &mg_dof);
```

Comment:
@kronbichler would you be amicable to this solution which avoids any code duplication and shall not break anything either? The whole transfer of solution seems to be working fine now even with several MPI cores.